### PR TITLE
Deployment template (for Windows based VMs) of Build and Release VSTS agent updated to .NET Core based ones.

### PR DESCRIPTION
**Pull request by lukka**
### Contributing guide

https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md
### Changelog
-  Windows based VM image based on VS2013, VS2015 or VS Dev15;
-  Updated Azure SDK  to 2.9
-  .Net Core based Agent of Build and Release (always latest release);
- Renamed of files/dirs./variables from VSO to VSTS
### Description of the change

Update the deployment template to use the most recent version of: 
- OS image of Windows, 
- Visual Studio version;
- Azure SDK
- binaries of build agents.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@lukka__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (lucappa). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by lukka** [comment]
Fixed all the merging conflicts.

Note to replace the external PS script URI in the azure.json file, i.e.:

```
"fileUris": [
"https://raw.githubusercontent.com/lukka/azure-quickstart-templates/master/visual-studio-vstsbuildagent-vm/InstallVSTSAgent.ps1"
],
```

must be replaced with:

```
"fileUris": [
"https://raw.githubusercontent.com/microsoft/azure-quickstart-templates/master/visual-studio-vstsbuildagent-vm/InstallVSTSAgent.ps1"
],
```

File: visual-studio-vstsbuildagent-vm/azuredeploy.json [hunk 1]
```diff
@@ -67,53 +70,46 @@
...
-    "vmSubnetRef": "[concat(variables('vmVnetID'), '/subnets/', variables('vnet01Subnet1Name'))]",
+    "vnet01Prefix": "10.0.0.0/16",
+    "vnet01Subnet1Name": "Subnet-1",
+    "vnet01Subnet1Prefix": "10.0.0.0/24",
     "vmStorageAccountContainerName": "vhds",
     "vmNicName": "[concat(parameters('vmName'), 'NetworkInterface')]",
-    "vmIP01Name": "VMIP01"
+    "vmIP01Name": "VMIP01",
+    "vmVnetID": "[resourceId('Microsoft.Network/virtualNetworks', 'Vnet01')]",
+    "vmSubnetRef": "[concat(variables('VMVnetID'), '/subnets/', variables('Vnet01Subnet1Name'))]"
```
Comment (bmoore-msft) [inline visual-studio-vstsbuildagent-vm/azuredeploy.json]: Just use: resourceId('Microsoft.Network/virtualNetworks/subnets', 'Vnet01', variables('Vnet01Subnet1Name')

File: visual-studio-vstsbuildagent-vm/azuredeploy.json [hunk 2]
```diff
@@ -67,53 +70,46 @@
...
   },
   "resources": [
     {
       "name": "[variables('storageAccountName')]",
       "type": "Microsoft.Storage/storageAccounts",
       "location": "[resourceGroup().location]",
       "apiVersion": "2015-05-01-preview",
```
Comment (bmoore-msft) [inline visual-studio-vstsbuildagent-vm/azuredeploy.json]: update apiVersion - 2016-01-01

File: visual-studio-vstsbuildagent-vm/azuredeploy.json [hunk 3]
```diff
@@ -235,9 +231,9 @@
             "typeHandlerVersion": "1.4",
             "settings": {
               "fileUris": [
-                "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/visual-studio-vsobuildagent-vm/InstallVSOAgent.ps1"
+                "https://raw.githubusercontent.com/lukka/azure-quickstart-templates/master/visual-studio-vstsbuildagent-vm/InstallVSTSAgent.ps1"
```
Comment (bmoore-msft) [inline visual-studio-vstsbuildagent-vm/azuredeploy.json]: see bp checklist for using artifacts

**Comment by lukka** [comment]
Hello @bmoore-msft  as requested:
-simplified the reference to the subnet of the VNet;
-updated the API versions of the resources;
-applied best practices for artifacts location.

File: visual-studio-vstsbuildagent-vm/azuredeploy.parameters.json [hunk 4]
```diff
@@ -31,6 +31,9 @@
     },
     "poolName": {
       "value": "GEN-UNIQUE"
+    },
+    "_artifactsLocation": {
```
Comment (bmoore-msft) [inline visual-studio-vstsbuildagent-vm/azuredeploy.parameters.json]: remove this from the param file since this has a defaultValue set...

**Comment by lukka** [comment]
this pull request can be closed, efforts wasted :( another pull request has been integrated instead.

**Pull request closed without merging**